### PR TITLE
update collection extension specification

### PR DESCRIPTION
**Pull request by matthewhanson**
The current version of the collection spec was not the latest version in the original PR. As discussed at FOSS4G-NA @cholmes this may be because you weren't clear on the merits of the final version, or something with the merge may have happened.

The main difference here is that now a Collection is not defined as a specific set of metadata. The spec now includes some additional justification as to why this is a useful feature. In short, allowing data providers to define the common set of metadata themselves rather than relying on requirements on what combines a set of data providers maximum flexibility for what collections can be used for.

Maybe it's data from:

- a single satellite
- multiple satellites but with the same model sensor
- data from multiple sensors on a single satellite
- different satellites that have different data use licenses
- a tasking satellite vs a nadir viewing satellite
- a harmonized dataset that combines different platforms with different bands
- a sensor with a continuously variable filter (i.e., no fixed set of bands)

some metadata fields *do* make more sense as a collection, and these could be listed as such under the spec or extension. For example the EO extension should list the fields that are typically common in a dataset similar to Landsat or Sentinel.

### CODE REVIEW

**Comment by cholmes** [comment]
Changed to merge to dev, so we can stage a release there.

File: extensions/stac-collection-spec.md [hunk 1]
```diff
@@ -1,28 +1,24 @@
...
+If a metadata field is specified in `Collection`, it will be ignored in any `Item` that links to that `Collection`. This is important because a `Collection` is the metadta that is common across all `Item` objects. If a field is variable at all, it should be part of the `Item` record.
 
 #### Linking to a `Collection`
-An `Item` links to a collection by providing a "link" with "rel" set to "collection". For example
+An `Item` specifies the collection it belongs to in two places. One is a field called 'collection' which is a key to a `Collection` record (that also has a 'collection' keyword). This acts as a Foreign Key in the `Item` that links to a `Collection`. 
+
+An `Item` also provides a link to a collection under the links dictionary:
 ```
-{ "rel": "collection", "href": "link/to/collection/record" }
+{ "collection": { "href": "link/to/collection/record" }
```
[the remark this answers is not in the document]
Comment (matthewhanson) [reply extensions/stac-collection-spec.md]: @cholmes I see, you are saying the 'collection' isn't actually the collection data, it's more like a collection definition...the set of data that has these properties.

If a catalog existed we'd probably want to put a link for 'catalog' in there that points to that.
```
"links": {
  "catalog": {
    "rel": "catalog",
    "href": "link/to/catalog.json"
  }
}
```

So isn't the actual collection of items what we are basically calling a catalog? In that case I think it's fine to say this is a collection - we are simply defining a 'collection' as a parent set of metadata, and a 'catalog' is the set of items that is a member of that collection.

**Comment by matthewhanson** [comment]
@cholmes think I resolved the comments you had

File: extensions/stac-collection-spec.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cholmes) [reply extensions/stac-collection-spec.md]: Yeah, I think the collection of items is basically a catalog. But we have this catalog.json file in the static catalog that I think we should probably combine with the collection metadata.

File: extensions/stac-collection-spec.md [hunk 2]
```diff
@@ -1,28 +1,29 @@
...
 |----------------------|---------------------------|-------------------------|---------------------------------------------------------------------------------------------| 
-| collection_name | string | Collection Name | A name given to the Collection
-| collection_description | string (optional) | Collection Description | A human readable description of the collection
+| collection | string | Collection Name | A name given to the Collection
+| description | string (optional) | Collection Description | A human readable description of the collection
+
+A `Collection` does not have many specific fields, as it may contain any fields that are in the core spec as well as any other extension. This provides maximum flexibility to data providers, as some the set of common metadata fields can vary between different types of data. For instance, Landsat and Sentinel data always has a eo:off_nadir value of 0, because those satellites are always pointed downward (i.e., nadir), while satellite that can be pointed will have varying eo:off_nadir values.
```
Comment (fredliporace) [inline extensions/stac-collection-spec.md]: Landsat-8 may acquire off-nadir images for special collections, see:
https://landsat.usgs.gov/landsat-8-l8-data-users-handbook-section-2
"but special collections may be scheduled off-nadir." 
It seems to me that eo:plaform or eo:collections are better to be used as examples of fields that do not change.

**Pull request merged**